cql/ManageRareAbnormality.cql

```diff
@@ -627,6 +630,70 @@ define HistologyInterpretedAsCin3:
         aC in Dash."Histologic CIN3"
   )
 
+define AnyHistologyInterpretedAsCin3:
+  AnyTrue(
+    (Collate.SortedBiopsyReports.allConclusions) aC
+      return
+        aC in Dash."Histologic CIN3"
+  )
+
+define AnyHistologyInterpretedAsCin2:
+  AnyTrue(
+    (Collate.SortedBiopsyReports.allConclusions) aC
+      return
+        aC ~ Dash."CIN2"
+  )
+
+define MostSevereHistologyInterpretedAsCin2InPast3Years:
+  AnyTrue(
+    (Collate.SortedBiopsyReports.allConclusions) aC
+      return
+        aC ~ Dash."CIN2"
+  )
+
+define MostRecentCin2BiopsyInPast3Years:
```

Comment:
I believe this should be named "Any" instead of "MostRecent".

Reply:
It is using the SortedBiopsyReports list, ordered newest to oldest  and Coalesce, so the first match returned will be the most recent biopsy report. Same logic used for other "MostRecentXXXInPastZYears" definitions.

Reply:
You are filtering the SortedBiopsyReports list by only those reports that have a conclusionCode of CIN2. So the first match returned will not be the most recent biopsy report. It will be the most recent report that has a value of CIN2. This expression is doing the same thing as `AnyHistologyInterpretedAsCin2`, except adding a date range, and taking the most recent of the CIN2 results instead of returning `true`.

If the patient had a CIN2 and then CIN1 afterwards, this expression would still return the CIN2, even if it was not the most recent biopsy.

Compare this to "[HistologyInterpretedAsCin2](https://github.com/ccsm-cds-tools/ccsm-cds-with-tests/blob/cin2-surveillance/cql/ManageRareAbnormality.cql#L620)", which looks for the most recent biopsy, and then evaluates whether or not that biopsy is CIN2. _Technically_ the name of this expression is not incorrect, because we are looking for the most recent CIN2 biopsy here. But I think now that we have expressions that look for _any_ biopsy over a given span, as well as expressions that look for the _most recent_ biopsy over a given span, we need to be clear and consistent with how we label most recent. And if I see an expression titled "Most Recent..." I am thinking that it is looking at the most recent biopsy that a patient had, regardless of result.

Reply:
Mike came up with the labelling of "LastKnown______" to define the pattern here, where we are looking for the most recent biopsy out of the list of biopsies that are specifically resulted as "CIN2". This has been updated throughout PR 120, in commit https://github.com/ccsm-cds-tools/ccsm-cds-with-tests/pull/120/files#diff-79dcbe5d6961c9c3f6bda93de156d7f281b67a45deecf191662c1ac8cb1c9579. I like this approach!